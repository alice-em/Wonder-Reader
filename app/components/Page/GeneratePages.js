import React, { memo } from 'react';

import Context from '../../context/ContextFactory';
import Page from './Page';

const generateTotalSize = encodedPages => encodedPages.reduce((a, { width }) => a + width, 0);

const GeneratePages = () => (
  <Context.Consumer>
    {({ state: { encodedPages }}) => (
      encodedPages.map(({ key, page, width }) => (
        <Page
          key={key}
          id={`page${key}`}
          width={(width / generateTotalSize(encodedPages)) * 100}
          alt="comic page"
          src={page}
        />
      ))
    )}
  </Context.Consumer>
);

export { GeneratePages };
export default memo(GeneratePages);
