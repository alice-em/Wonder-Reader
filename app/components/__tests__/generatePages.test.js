import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';

import GeneratePages from '../generatePages';

Enzyme.configure({ adapter: new Adapter() });

const encodedPages = [
  {
    key: 0,
    page: 'abc.png',
    width: 100,
  },
  {
    key: 1,
    page: 'deb.png',
    width: 300,
  },
];
// const sampleData = [1, 2, 3, 4, 5].map(n => ({ width: n }));

describe('generatePages', () => {
  it('should render', () => {
    const wrapper = shallow(<GeneratePages encodedPages={encodedPages} />);
    expect(wrapper).toMatchSnapshot();
  });
});
