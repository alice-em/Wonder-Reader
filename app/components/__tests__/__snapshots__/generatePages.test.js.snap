// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generatePages should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <generatePages
    encodedPages={
      Array [
        Object {
          "key": 0,
          "page": "abc.png",
          "width": 100,
        },
        Object {
          "key": 1,
          "page": "deb.png",
          "width": 300,
        },
      ]
    }
    pages={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Page
      alt="comic page"
      id="page0"
      src="abc.png"
      width={25}
    />,
    <Page
      alt="comic page"
      id="page1"
      src="deb.png"
      width={75}
    />,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Page
        alt="comic page"
        id="page0"
        src="abc.png"
        width={25}
      />,
      <Page
        alt="comic page"
        id="page1"
        src="deb.png"
        width={75}
      />,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
